### ISSUE TITLE
Add heights to every gRPC request

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#149]
## Summary

Heights are not included in gRPC requests as the height can be set in the SDK context. This isn't very useful for non-go implementations which might have a difficult time setting the SDK context correctly. 

We should add a height request into the gRPC requests. `ibc-rs` is currently having difficulty doing gRPC requests at specific heights. 

The height in the request seems sufficiently generalized that other implementations could implement the same gRPC routes with the same proto files 

cc @fedekunze 

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by colin-axner** [comment]
[ref](https://github.com/cosmos/ibc/issues/568)

**Comment by noot** [comment]
@colin-axner any progress/updates on this? I'm working on integrating IBC in Rust into Astria and have run into this as a major issue, since I would rather use only gRPC requests rather than the ABCI + gRPC combo. I'm happy to take this issue on!

**Comment by DimitrisJim** [comment]
doesn't appear that this was closed due to some blocker so don't see why you couldn't take it on and push it through, rationale still appears to hold :smile:. cc @crodriguezvega for a second opinion.

**Comment by crodriguezvega** [comment]
No, this hasn't been implemented and there hasn't been any progress, so thanks @noot if you want to work on it. For reference, [here's a PR](https://github.com/cosmos/ibc-go/pull/248) where Colin started to work on this but eventually closed.

**Comment by noot** [comment]
great, thanks @DimitrisJim and @crodriguezvega ! I'll take this on then :)

**Comment by crodriguezvega** [comment]
I am going to tentatively put this in v11...

**Comment by gjermundgaraba** [comment]
I'll close this for now. While this might be useful, it is not going to be prioritized unless there is a clear need for it right now. We can reopen and reconsider in the future if needed.
